clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -69,19 +72,33 @@
             this.transport = transport;
         }
 
+        if (handshakeResponseTimeout != null) {
+            this.handshakeResponseTimeout = handshakeResponseTimeout;
+        }
+
         this.skipNegotiate = skipNegotiate;
 
         this.callback = (payload) -> {
             if (!handshakeReceived) {
                 int handshakeLength = payload.indexOf(RECORD_SEPARATOR) + 1;
                 String handshakeResponseString = payload.substring(0, handshakeLength - 1);
-                HandshakeResponseMessage handshakeResponse = HandshakeProtocol.parseHandshakeResponse(handshakeResponseString);
+                HandshakeResponseMessage handshakeResponse;
+                try {
+                    handshakeResponse = HandshakeProtocol.parseHandshakeResponse(handshakeResponseString);
+                } catch (RuntimeException ex) {
+                    RuntimeException exception = new RuntimeException("An invalid handshake response was received from the server.");
```

Comment:
Don't swallow the exception: `new RuntimeException("...", ex);`

Reply:
Meant to do that. Good catch